### PR TITLE
mktemp: replace '@' with '-' in temp path

### PR DESCRIPTION
**Pull request by jnooree**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?
- [x] Have you successfully run `brew man` locally and committed any changes?

-----

Some formula (e.g. [gcc@10](https://github.com/Homebrew/linuxbrew-core/blob/master/Formula/gcc@10.rb)) fails to build if the temp directory contains an at sign (`@`). This pull request replace the character into a hyphen (`-`) for temp directories to avoid build failures.

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
> Some formula (e.g. [gcc@10](https://github.com/Homebrew/linuxbrew-core/blob/master/Formula/gcc@10.rb)) fails to build if the temp directory contains an at sign (`@`). This pull request replace the character into a hyphen (`-`) for temp directories to avoid build failures.

In which circumstances does this happen (e.g. `brew config` output)?

File: Library/Homebrew/mktemp.rb [hunk 1]
```diff
@@ -35,7 +35,7 @@ def to_s
   end
 
   def run
-    @tmpdir = Pathname.new(Dir.mktmpdir("#{@prefix}-", HOMEBREW_TEMP))
+    @tmpdir = Pathname.new(Dir.mktmpdir("#{@prefix.dup.tr! "@", "-"}-", HOMEBREW_TEMP))
```
Comment (MikeMcQuaid) [inline Library/Homebrew/mktemp.rb:38]:
```suggestion
    @tmpdir = Pathname.new(Dir.mktmpdir("#{@prefix.tr "@", "-"}-", HOMEBREW_TEMP))
```

But hold off until we decide whether we're going to merge this in any form, thanks!

Comment (jnooree) [reply Library/Homebrew/mktemp.rb:38]: If change `dup.tr` into `tr`, then the original `@prefix` will be modified also, so homebrew will try to install at `$HOMEBREW_PREFIX/Cellar/gcc-10` (for example).

**Comment by jnooree** [comment]
> > Some formula (e.g. [gcc@10](https://github.com/Homebrew/linuxbrew-core/blob/master/Formula/gcc@10.rb)) fails to build if the temp directory contains an at sign (`@`). This pull request replace the character into a hyphen (`-`) for temp directories to avoid build failures.
> 
> In which circumstances does this happen (e.g. `brew config` output)?

It was on Ubuntu Linux LTS 16.04, with the default settings of the homebrew.

File: Library/Homebrew/mktemp.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/mktemp.rb:38]: Nope, `.tr!` will change the original `@prefix` but `.tr` will return a new string.

**Comment by MikeMcQuaid** [comment]
@jnooree Can I see your `brew doctor`, `mount -a` and `brew ruby -e 'puts HOMEBREW_TEMP'` output, too? Thanks!

File: Library/Homebrew/mktemp.rb (hunk 1 above)
Comment (jnooree) [reply Library/Homebrew/mktemp.rb:38]: Oh ok, thanks for the fix!

**Comment by jnooree** [comment]
> @jnooree Can I see your `brew doctor`, `mount -a` and `brew ruby -e 'puts HOMEBREW_TEMP'` output, too? Thanks!

`brew doctor`:

```
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /opt/python/current/bin/python2-config
  /opt/python/current/bin/python-config
  /opt/python/current/bin/python2.7-config
  /opt/python/3.8.5/bin/python3.8-config
  /opt/python/3.8.5/bin/python3-config

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/boost/container/detail/alloc_lib.h
  /usr/local/include/boost/predef.h
  /usr/local/include/boost/predef/architecture.h
  /usr/local/include/boost/predef/architecture/alpha.h
  /usr/local/include/boost/predef/architecture/arm.h
  /usr/local/include/boost/predef/architecture/blackfin.h
  /usr/local/include/boost/predef/architecture/convex.h
  /usr/local/include/boost/predef/architecture/ia64.h
  /usr/local/include/boost/predef/architecture/m68k.h
  /usr/local/include/boost/predef/architecture/mips.h
  /usr/local/include/boost/predef/architecture/parisc.h
  /usr/local/include/boost/predef/architecture/ppc.h
  /usr/local/include/boost/predef/architecture/pyramid.h
  /usr/local/include/boost/predef/architecture/rs6k.h
  /usr/local/include/boost/predef/architecture/sparc.h
  /usr/local/include/boost/predef/architecture/superh.h
  /usr/local/include/boost/predef/architecture/sys370.h
  /usr/local/include/boost/predef/architecture/sys390.h
  /usr/local/include/boost/predef/architecture/x86.h
  /usr/local/include/boost/predef/architecture/x86/32.h
  /usr/local/include/boost/predef/architecture/x86/64.h
  /usr/local/include/boost/predef/architecture/z.h
  /usr/local/include/boost/predef/compiler.h
  /usr/local/include/boost/predef/compiler/borland.h
  /usr/local/include/boost/predef/compiler/clang.h
  /usr/local/include/boost/predef/compiler/comeau.h
  /usr/local/include/boost/predef/compiler/compaq.h
  /usr/local/include/boost/predef/compiler/diab.h
  /usr/local/include/boost/predef/compiler/digitalmars.h
  /usr/local/include/boost/predef/compiler/dignus.h
  /usr/local/include/boost/predef/compiler/edg.h
  /usr/local/include/boost/predef/compiler/ekopath.h
  /usr/local/include/boost/predef/compiler/gcc.h
  /usr/local/include/boost/predef/compiler/gcc_xml.h
  /usr/local/include/boost/predef/compiler/greenhills.h
  /usr/local/include/boost/predef/compiler/hp_acc.h
  /usr/local/include/boost/predef/compiler/iar.h
  /usr/local/include/boost/predef/compiler/ibm.h
  /usr/local/include/boost/predef/compiler/intel.h
  /usr/local/include/boost/predef/compiler/kai.h
  /usr/local/include/boost/predef/compiler/llvm.h
  /usr/local/include/boost/predef/compiler/metaware.h
  /usr/local/include/boost/predef/compiler/metrowerks.h
  /usr/local/include/boost/predef/compiler/microtec.h
  /usr/local/include/boost/predef/compiler/mpw.h
  /usr/local/include/boost/predef/compiler/palm.h
  /usr/local/include/boost/predef/compiler/pgi.h
  /usr/local/include/boost/predef/compiler/sgi_mipspro.h
  /usr/local/include/boost/predef/compiler/sunpro.h
  /usr/local/include/boost/predef/compiler/tendra.h
  /usr/local/include/boost/predef/compiler/visualc.h
  /usr/local/include/boost/predef/compiler/watcom.h
  /usr/local/include/boost/predef/detail/_cassert.h
  /usr/local/include/boost/predef/detail/_exception.h
  /usr/local/include/boost/predef/detail/comp_detected.h
  /usr/local/include/boost/predef/detail/endian_compat.h
  /usr/local/include/boost/predef/detail/os_detected.h
  /usr/local/include/boost/predef/detail/platform_detected.h
  /usr/local/include/boost/predef/detail/test.h
  /usr/local/include/boost/predef/detail/test_def.h
  /usr/local/include/boost/predef/hardware.h
  /usr/local/include/boost/predef/hardware/simd.h
  /usr/local/include/boost/predef/hardware/simd/arm.h
  /usr/local/include/boost/predef/hardware/simd/arm/versions.h
  /usr/local/include/boost/predef/hardware/simd/ppc.h
  /usr/local/include/boost/predef/hardware/simd/ppc/versions.h
  /usr/local/include/boost/predef/hardware/simd/x86.h
  /usr/local/include/boost/predef/hardware/simd/x86/versions.h
  /usr/local/include/boost/predef/hardware/simd/x86_amd.h
  /usr/local/include/boost/predef/hardware/simd/x86_amd/versions.h
  /usr/local/include/boost/predef/language.h
  /usr/local/include/boost/predef/language/objc.h
  /usr/local/include/boost/predef/language/stdc.h
  /usr/local/include/boost/predef/language/stdcpp.h
  /usr/local/include/boost/predef/library.h
  /usr/local/include/boost/predef/library/c.h
  /usr/local/include/boost/predef/library/c/_prefix.h
  /usr/local/include/boost/predef/library/c/cloudabi.h
  /usr/local/include/boost/predef/library/c/gnu.h
  /usr/local/include/boost/predef/library/c/uc.h
  /usr/local/include/boost/predef/library/c/vms.h
  /usr/local/include/boost/predef/library/c/zos.h
  /usr/local/include/boost/predef/library/std.h
  /usr/local/include/boost/predef/library/std/_prefix.h
  /usr/local/include/boost/predef/library/std/cxx.h
  /usr/local/include/boost/predef/library/std/dinkumware.h
  /usr/local/include/boost/predef/library/std/libcomo.h
  /usr/local/include/boost/predef/library/std/modena.h
  /usr/local/include/boost/predef/library/std/msl.h
  /usr/local/include/boost/predef/library/std/roguewave.h
  /usr/local/include/boost/predef/library/std/sgi.h
  /usr/local/include/boost/predef/library/std/stdcpp3.h
  /usr/local/include/boost/predef/library/std/stlport.h
  /usr/local/include/boost/predef/library/std/vacpp.h
  /usr/local/include/boost/predef/make.h
  /usr/local/include/boost/predef/os.h
  /usr/local/include/boost/predef/os/aix.h
  /usr/local/include/boost/predef/os/amigaos.h
  /usr/local/include/boost/predef/os/android.h
  /usr/local/include/boost/predef/os/beos.h
  /usr/local/include/boost/predef/os/bsd.h
  /usr/local/include/boost/predef/os/bsd/bsdi.h
  /usr/local/include/boost/predef/os/bsd/dragonfly.h
  /usr/local/include/boost/predef/os/bsd/free.h
  /usr/local/include/boost/predef/os/bsd/net.h
  /usr/local/include/boost/predef/os/bsd/open.h
  /usr/local/include/boost/predef/os/cygwin.h
  /usr/local/include/boost/predef/os/haiku.h
  /usr/local/include/boost/predef/os/hpux.h
  /usr/local/include/boost/predef/os/ios.h
  /usr/local/include/boost/predef/os/irix.h
  /usr/local/include/boost/predef/os/linux.h
  /usr/local/include/boost/predef/os/macos.h
  /usr/local/include/boost/predef/os/os400.h
  /usr/local/include/boost/predef/os/qnxnto.h
  /usr/local/include/boost/predef/os/solaris.h
  /usr/local/include/boost/predef/os/unix.h
  /usr/local/include/boost/predef/os/vms.h
  /usr/local/include/boost/predef/os/windows.h
  /usr/local/include/boost/predef/other.h
  /usr/local/include/boost/predef/other/endian.h
  /usr/local/include/boost/predef/other/workaround.h
  /usr/local/include/boost/predef/platform.h
  /usr/local/include/boost/predef/platform/cloudabi.h
  /usr/local/include/boost/predef/platform/ios.h
  /usr/local/include/boost/predef/platform/mingw.h
  /usr/local/include/boost/predef/platform/mingw32.h
  /usr/local/include/boost/predef/platform/mingw64.h
  /usr/local/include/boost/predef/platform/windows_desktop.h
  /usr/local/include/boost/predef/platform/windows_phone.h
  /usr/local/include/boost/predef/platform/windows_runtime.h
  /usr/local/include/boost/predef/platform/windows_server.h
  /usr/local/include/boost/predef/platform/windows_store.h
  /usr/local/include/boost/predef/platform/windows_system.h
  /usr/local/include/boost/predef/platform/windows_uwp.h
  /usr/local/include/boost/predef/version.h
  /usr/local/include/boost/predef/version_number.h
  /usr/local/include/boost/python/detail/python22_fixed.h
  /usr/local/include/boost/regex.h
  /usr/local/include/boost/stacktrace/detail/pop_options.h
  /usr/local/include/boost/stacktrace/detail/push_options.h
  /usr/local/include/ffindex.h
  /usr/local/include/ffutil.h
  /usr/local/include/google/sparsehash/densehashtable.h
  /usr/local/include/google/sparsehash/hashtable-common.h
  /usr/local/include/google/sparsehash/libc_allocator_with_realloc.h
  /usr/local/include/google/sparsehash/sparsehashtable.h
  /usr/local/include/google/template_util.h
  /usr/local/include/google/type_traits.h
  /usr/local/include/sparsehash/internal/densehashtable.h
  /usr/local/include/sparsehash/internal/hashtable-common.h
  /usr/local/include/sparsehash/internal/libc_allocator_with_realloc.h
  /usr/local/include/sparsehash/internal/sparseconfig.h
  /usr/local/include/sparsehash/internal/sparsehashtable.h
  /usr/local/include/sparsehash/template_util.h
  /usr/local/include/sparsehash/type_traits.h

Warning: Unbrewed .pc files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected .pc files:
  /usr/local/lib/pkgconfig/libsparsehash.pc

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libboost_atomic.a
  /usr/local/lib/libboost_chrono.a
  /usr/local/lib/libboost_container.a
  /usr/local/lib/libboost_context.a
  /usr/local/lib/libboost_coroutine.a
  /usr/local/lib/libboost_date_time.a
  /usr/local/lib/libboost_exception.a
  /usr/local/lib/libboost_filesystem.a
  /usr/local/lib/libboost_graph.a
  /usr/local/lib/libboost_iostreams.a
  /usr/local/lib/libboost_locale.a
  /usr/local/lib/libboost_log.a
  /usr/local/lib/libboost_log_setup.a
  /usr/local/lib/libboost_math_c99.a
  /usr/local/lib/libboost_math_c99f.a
  /usr/local/lib/libboost_math_c99l.a
  /usr/local/lib/libboost_math_tr1.a
  /usr/local/lib/libboost_math_tr1f.a
  /usr/local/lib/libboost_math_tr1l.a
  /usr/local/lib/libboost_numpy.a
  /usr/local/lib/libboost_prg_exec_monitor.a
  /usr/local/lib/libboost_program_options.a
  /usr/local/lib/libboost_python.a
  /usr/local/lib/libboost_random.a
  /usr/local/lib/libboost_regex.a
  /usr/local/lib/libboost_serialization.a
  /usr/local/lib/libboost_signals.a
  /usr/local/lib/libboost_stacktrace_addr2line.a
  /usr/local/lib/libboost_stacktrace_backtrace.a
  /usr/local/lib/libboost_stacktrace_basic.a
  /usr/local/lib/libboost_stacktrace_noop.a
  /usr/local/lib/libboost_system.a
  /usr/local/lib/libboost_test_exec_monitor.a
  /usr/local/lib/libboost_thread.a
  /usr/local/lib/libboost_timer.a
  /usr/local/lib/libboost_type_erasure.a
  /usr/local/lib/libboost_unit_test_framework.a
  /usr/local/lib/libboost_wave.a
  /usr/local/lib/libboost_wserialization.a

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  meson
  zsh
  rdkit
  numpy

Warning: You have uncommitted modifications to Homebrew/homebrew-core.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /home/jnooree/.linuxbrew/Library/Taps/homebrew/homebrew-core && git stash && git clean -d -f

Uncommitted files:
   M Formula/gdb.rb

Warning: Your Homebrew's prefix is not /home/linuxbrew/.linuxbrew.
Some of Homebrew's bottles (binary packages) can only be used with the default
prefix (/home/linuxbrew/.linuxbrew).
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Discourse, Twitter or IRC. You are responsible for resolving any issues you
experience while you are running this unsupported configuration.

[/home/jnooree] less doctor.txt                                                 @galaxy2
[/home/jnooree] cat doctor.txt                                                  @galaxy2
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /opt/python/current/bin/python2-config
  /opt/python/current/bin/python-config
  /opt/python/current/bin/python2.7-config
  /opt/python/3.8.5/bin/python3.8-config
  /opt/python/3.8.5/bin/python3-config

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/boost/container/detail/alloc_lib.h
  /usr/local/include/boost/predef.h
  /usr/local/include/boost/predef/architecture.h
  /usr/local/include/boost/predef/architecture/alpha.h
  /usr/local/include/boost/predef/architecture/arm.h
  /usr/local/include/boost/predef/architecture/blackfin.h
  /usr/local/include/boost/predef/architecture/convex.h
  /usr/local/include/boost/predef/architecture/ia64.h
  /usr/local/include/boost/predef/architecture/m68k.h
  /usr/local/include/boost/predef/architecture/mips.h
  /usr/local/include/boost/predef/architecture/parisc.h
  /usr/local/include/boost/predef/architecture/ppc.h
  /usr/local/include/boost/predef/architecture/pyramid.h
  /usr/local/include/boost/predef/architecture/rs6k.h
  /usr/local/include/boost/predef/architecture/sparc.h
  /usr/local/include/boost/predef/architecture/superh.h
  /usr/local/include/boost/predef/architecture/sys370.h
  /usr/local/include/boost/predef/architecture/sys390.h
  /usr/local/include/boost/predef/architecture/x86.h
  /usr/local/include/boost/predef/architecture/x86/32.h
  /usr/local/include/boost/predef/architecture/x86/64.h
  /usr/local/include/boost/predef/architecture/z.h
  /usr/local/include/boost/predef/compiler.h
  /usr/local/include/boost/predef/compiler/borland.h
  /usr/local/include/boost/predef/compiler/clang.h
  /usr/local/include/boost/predef/compiler/comeau.h
  /usr/local/include/boost/predef/compiler/compaq.h
  /usr/local/include/boost/predef/compiler/diab.h
  /usr/local/include/boost/predef/compiler/digitalmars.h
  /usr/local/include/boost/predef/compiler/dignus.h
  /usr/local/include/boost/predef/compiler/edg.h
  /usr/local/include/boost/predef/compiler/ekopath.h
  /usr/local/include/boost/predef/compiler/gcc.h
  /usr/local/include/boost/predef/compiler/gcc_xml.h
  /usr/local/include/boost/predef/compiler/greenhills.h
  /usr/local/include/boost/predef/compiler/hp_acc.h
  /usr/local/include/boost/predef/compiler/iar.h
  /usr/local/include/boost/predef/compiler/ibm.h
  /usr/local/include/boost/predef/compiler/intel.h
  /usr/local/include/boost/predef/compiler/kai.h
  /usr/local/include/boost/predef/compiler/llvm.h
  /usr/local/include/boost/predef/compiler/metaware.h
  /usr/local/include/boost/predef/compiler/metrowerks.h
  /usr/local/include/boost/predef/compiler/microtec.h
  /usr/local/include/boost/predef/compiler/mpw.h
  /usr/local/include/boost/predef/compiler/palm.h
  /usr/local/include/boost/predef/compiler/pgi.h
  /usr/local/include/boost/predef/compiler/sgi_mipspro.h
  /usr/local/include/boost/predef/compiler/sunpro.h
  /usr/local/include/boost/predef/compiler/tendra.h
  /usr/local/include/boost/predef/compiler/visualc.h
  /usr/local/include/boost/predef/compiler/watcom.h
  /usr/local/include/boost/predef/detail/_cassert.h
  /usr/local/include/boost/predef/detail/_exception.h
  /usr/local/include/boost/predef/detail/comp_detected.h
  /usr/local/include/boost/predef/detail/endian_compat.h
  /usr/local/include/boost/predef/detail/os_detected.h
  /usr/local/include/boost/predef/detail/platform_detected.h
  /usr/local/include/boost/predef/detail/test.h
  /usr/local/include/boost/predef/detail/test_def.h
  /usr/local/include/boost/predef/hardware.h
  /usr/local/include/boost/predef/hardware/simd.h
  /usr/local/include/boost/predef/hardware/simd/arm.h
  /usr/local/include/boost/predef/hardware/simd/arm/versions.h
  /usr/local/include/boost/predef/hardware/simd/ppc.h
  /usr/local/include/boost/predef/hardware/simd/ppc/versions.h
  /usr/local/include/boost/predef/hardware/simd/x86.h
  /usr/local/include/boost/predef/hardware/simd/x86/versions.h
  /usr/local/include/boost/predef/hardware/simd/x86_amd.h
  /usr/local/include/boost/predef/hardware/simd/x86_amd/versions.h
  /usr/local/include/boost/predef/language.h
  /usr/local/include/boost/predef/language/objc.h
  /usr/local/include/boost/predef/language/stdc.h
  /usr/local/include/boost/predef/language/stdcpp.h
  /usr/local/include/boost/predef/library.h
  /usr/local/include/boost/predef/library/c.h
  /usr/local/include/boost/predef/library/c/_prefix.h
  /usr/local/include/boost/predef/library/c/cloudabi.h
  /usr/local/include/boost/predef/library/c/gnu.h
  /usr/local/include/boost/predef/library/c/uc.h
  /usr/local/include/boost/predef/library/c/vms.h
  /usr/local/include/boost/predef/library/c/zos.h
  /usr/local/include/boost/predef/library/std.h
  /usr/local/include/boost/predef/library/std/_prefix.h
  /usr/local/include/boost/predef/library/std/cxx.h
  /usr/local/include/boost/predef/library/std/dinkumware.h
  /usr/local/include/boost/predef/library/std/libcomo.h
  /usr/local/include/boost/predef/library/std/modena.h
  /usr/local/include/boost/predef/library/std/msl.h
  /usr/local/include/boost/predef/library/std/roguewave.h
  /usr/local/include/boost/predef/library/std/sgi.h
  /usr/local/include/boost/predef/library/std/stdcpp3.h
  /usr/local/include/boost/predef/library/std/stlport.h
  /usr/local/include/boost/predef/library/std/vacpp.h
  /usr/local/include/boost/predef/make.h
  /usr/local/include/boost/predef/os.h
  /usr/local/include/boost/predef/os/aix.h
  /usr/local/include/boost/predef/os/amigaos.h
  /usr/local/include/boost/predef/os/android.h
  /usr/local/include/boost/predef/os/beos.h
  /usr/local/include/boost/predef/os/bsd.h
  /usr/local/include/boost/predef/os/bsd/bsdi.h
  /usr/local/include/boost/predef/os/bsd/dragonfly.h
  /usr/local/include/boost/predef/os/bsd/free.h
  /usr/local/include/boost/predef/os/bsd/net.h
  /usr/local/include/boost/predef/os/bsd/open.h
  /usr/local/include/boost/predef/os/cygwin.h
  /usr/local/include/boost/predef/os/haiku.h
  /usr/local/include/boost/predef/os/hpux.h
  /usr/local/include/boost/predef/os/ios.h
  /usr/local/include/boost/predef/os/irix.h
  /usr/local/include/boost/predef/os/linux.h
  /usr/local/include/boost/predef/os/macos.h
  /usr/local/include/boost/predef/os/os400.h
  /usr/local/include/boost/predef/os/qnxnto.h
  /usr/local/include/boost/predef/os/solaris.h
  /usr/local/include/boost/predef/os/unix.h
  /usr/local/include/boost/predef/os/vms.h
  /usr/local/include/boost/predef/os/windows.h
  /usr/local/include/boost/predef/other.h
  /usr/local/include/boost/predef/other/endian.h
  /usr/local/include/boost/predef/other/workaround.h
  /usr/local/include/boost/predef/platform.h
  /usr/local/include/boost/predef/platform/cloudabi.h
  /usr/local/include/boost/predef/platform/ios.h
  /usr/local/include/boost/predef/platform/mingw.h
  /usr/local/include/boost/predef/platform/mingw32.h
  /usr/local/include/boost/predef/platform/mingw64.h
  /usr/local/include/boost/predef/platform/windows_desktop.h
  /usr/local/include/boost/predef/platform/windows_phone.h
  /usr/local/include/boost/predef/platform/windows_runtime.h
  /usr/local/include/boost/predef/platform/windows_server.h
  /usr/local/include/boost/predef/platform/windows_store.h
  /usr/local/include/boost/predef/platform/windows_system.h
  /usr/local/include/boost/predef/platform/windows_uwp.h
  /usr/local/include/boost/predef/version.h
  /usr/local/include/boost/predef/version_number.h
  /usr/local/include/boost/python/detail/python22_fixed.h
  /usr/local/include/boost/regex.h
  /usr/local/include/boost/stacktrace/detail/pop_options.h
  /usr/local/include/boost/stacktrace/detail/push_options.h
  /usr/local/include/ffindex.h
  /usr/local/include/ffutil.h
  /usr/local/include/google/sparsehash/densehashtable.h
  /usr/local/include/google/sparsehash/hashtable-common.h
  /usr/local/include/google/sparsehash/libc_allocator_with_realloc.h
  /usr/local/include/google/sparsehash/sparsehashtable.h
  /usr/local/include/google/template_util.h
  /usr/local/include/google/type_traits.h
  /usr/local/include/sparsehash/internal/densehashtable.h
  /usr/local/include/sparsehash/internal/hashtable-common.h
  /usr/local/include/sparsehash/internal/libc_allocator_with_realloc.h
  /usr/local/include/sparsehash/internal/sparseconfig.h
  /usr/local/include/sparsehash/internal/sparsehashtable.h
  /usr/local/include/sparsehash/template_util.h
  /usr/local/include/sparsehash/type_traits.h

Warning: Unbrewed .pc files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected .pc files:
  /usr/local/lib/pkgconfig/libsparsehash.pc

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libboost_atomic.a
  /usr/local/lib/libboost_chrono.a
  /usr/local/lib/libboost_container.a
  /usr/local/lib/libboost_context.a
  /usr/local/lib/libboost_coroutine.a
  /usr/local/lib/libboost_date_time.a
  /usr/local/lib/libboost_exception.a
  /usr/local/lib/libboost_filesystem.a
  /usr/local/lib/libboost_graph.a
  /usr/local/lib/libboost_iostreams.a
  /usr/local/lib/libboost_locale.a
  /usr/local/lib/libboost_log.a
  /usr/local/lib/libboost_log_setup.a
  /usr/local/lib/libboost_math_c99.a
  /usr/local/lib/libboost_math_c99f.a
  /usr/local/lib/libboost_math_c99l.a
  /usr/local/lib/libboost_math_tr1.a
  /usr/local/lib/libboost_math_tr1f.a
  /usr/local/lib/libboost_math_tr1l.a
  /usr/local/lib/libboost_numpy.a
  /usr/local/lib/libboost_prg_exec_monitor.a
  /usr/local/lib/libboost_program_options.a
  /usr/local/lib/libboost_python.a
  /usr/local/lib/libboost_random.a
  /usr/local/lib/libboost_regex.a
  /usr/local/lib/libboost_serialization.a
  /usr/local/lib/libboost_signals.a
  /usr/local/lib/libboost_stacktrace_addr2line.a
  /usr/local/lib/libboost_stacktrace_backtrace.a
  /usr/local/lib/libboost_stacktrace_basic.a
  /usr/local/lib/libboost_stacktrace_noop.a
  /usr/local/lib/libboost_system.a
  /usr/local/lib/libboost_test_exec_monitor.a
  /usr/local/lib/libboost_thread.a
  /usr/local/lib/libboost_timer.a
  /usr/local/lib/libboost_type_erasure.a
  /usr/local/lib/libboost_unit_test_framework.a
  /usr/local/lib/libboost_wave.a
  /usr/local/lib/libboost_wserialization.a

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  meson
  zsh
  rdkit
  numpy

Warning: You have uncommitted modifications to Homebrew/homebrew-core.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /home/jnooree/.linuxbrew/Library/Taps/homebrew/homebrew-core && git stash && git clean -d -f

Uncommitted files:
   M Formula/gdb.rb

Warning: Your Homebrew's prefix is not /home/linuxbrew/.linuxbrew.
Some of Homebrew's bottles (binary packages) can only be used with the default
prefix (/home/linuxbrew/.linuxbrew).
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Discourse, Twitter or IRC. You are responsible for resolving any issues you
experience while you are running this unsupported configuration.
```

`mount -a` has no output, and `brew ruby -e 'puts HOMEBREW_TEMP'` prints `/tmp`.

**Comment by MikeMcQuaid** [comment]
What filesystem is in use in `/tmp` and what distro are you on?

**Comment by jnooree** [comment]
It is `ext4`, and for the distro,

> It was on Ubuntu Linux LTS 16.04, with the default settings of the homebrew.

**Comment by MikeMcQuaid** [comment]
CC @sjackman and @maxim-belkin for help here. There's definitely something weird going on as we'd otherwise have a lot more reports of this than we do(n't).

**Comment by maxim-belkin** [comment]
I'd appreciate a `gist-log` of a failed build.

**Comment by jnooree** [comment]
@maxim-belkin 

For default settings: https://gist.github.com/jnooree/94b73fdc5867024a4293a419dce0ba8e
For gcc-9 and g++-9 as compilers: https://gist.github.com/jnooree/bcc8afc957f51566d6a1344f2ba0b119

same error for the two cases.

**Comment by maxim-belkin** [comment]
> P.S. If the homebrew output with `--debug --verbose` options will be helpful, I can also attach those.

There is no need for that, thank you. What you attached is sufficient.

So, what happens is that temporary directory name with an `@` gets into some Texinfo (documentation) files where `@` symbol is treated specially (as a command), which causes failures like:

```
../../gcc/doc/invoke.texi:1851: unknown command `10-20201031-9668-184lfzj'
../../gcc/doc/invoke.texi:1851: @include: could not find /tmp/gcc/gcc-10.2.0/build/gcc/../../gcc/../libiberty/at-file.texi
```

So, in that sense, I'm curious as to why this didn't happen (or did it?) for @jonchang when he built the Linux bottle (commit Homebrew/homebrew-core@b430e1e7826). @jonchang, any comments?

Anyways, I think this is an upstream (gcc) bug as Homebrew isn't doing anything super-wrong here. I'd suggest you (@jnooree) report this to GCC, see what they say. In the meantime, we could do a short-term patch until GCC fixes it but I'll leave that decision to others.

**Comment by jonchang** [comment]
Unfortunately the logs for that build have expired. But we definitely wouldn't have hit this bug because otherwise the bottles for gcc@10 wouldn't have built at all. My guess is that some dependency changed between now and then and now gcc@10 doesn't build from source. I kicked off a test build just to see what happens, so we'll see if the same bug occurs now: https://github.com/Homebrew/linuxbrew-core/runs/1335738978?check_suite_focus=true

**Comment by fxcoudert** [comment]
If I remember correctly, this issue is part of the reason why, in the versioned gcc formulas, we have:

```
    # Even when suffixes are appended, the info pages conflict when
    # install-info is run so pretend we have an outdated makeinfo
    # to prevent their build.
    ENV["gcc_cv_prog_makeinfo_modern"] = "no"
```

The comment gives one reason (the info pages conflict) but the second reason is because of this makeinfo bug.

**Comment by jnooree** [comment]
One thing I'm curious about: is it a problem changing the temporary build directory naming method that could potentially cause some errors? No offenses here, sorry if it sounds like (I'm not a native).

**Comment by maxim-belkin** [comment]
@fxcoudert, I don't see `gcc_cv_prog_makeinfo_modern` in gcc@10.

> is it a problem changing the temporary build directory naming method that could potentially cause some errors?

If that was a general problem - yes, sure. But in this case, I think upstream (gcc) should fix their scripts to allow `@` symbols in temporary directories' names.

**Comment by fxcoudert** [comment]
> I don't see gcc_cv_prog_makeinfo_modern in gcc@10

There is no `gcc@10`, only `gcc`

**Comment by fxcoudert** [comment]
Upstream GCC bug report is here: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=57076

**Comment by fxcoudert** [comment]
I've submitted a patch to GCC: https://gcc.gnu.org/pipermail/gcc-patches/2020-October/557681.html

**Comment by maxim-belkin** [comment]
> > I don't see gcc_cv_prog_makeinfo_modern in gcc@10
> 
> 
> 
> There is no `gcc@10`, only `gcc`

Sure there is: https://github.com/Homebrew/linuxbrew-core/blob/master/Formula/gcc@10.rb

**Comment by fxcoudert** [comment]
@maxim-belkin I meant in homebrew-core. I don't contribute to linuxbrew-core so I can't really have an idea about why it chose to do things differently.

**Comment by MikeMcQuaid** [comment]
Thanks all! Passing on this PR as-is as it seems we have multiple maintainers who agree this isn't the right direction. Thanks anyway @jnooree!

**Pull request closed without merging**